pkg/config/config_template.yaml

```diff
@@ -1588,3 +1588,79 @@ api_key:
   # key_file: ""
 
 {{ end -}}
+{{- if .SNMP }}
+
+########################
+## SNMP Configuration ##
+########################
+
+## @param snmp_traps_enabled - boolean - optional - default: false
+## Set this option to true to enable the SNMP traps server.
+#
+# snmp_traps_enabled: false
+
+## stop_timeout - float - optional - default: 5.0
+## The maximum number of seconds to wait for listeners to stop when the Agent shuts down.
+#
+# snmp_traps_stop_timeout: 5.0
+
+## @param snmp_traps_listeners - list of custom objects - optional
+## Configure SNMP traps listeners. Each listener is associated to a UDP port and a set of SNMP credentials.
+#
+# snmp_traps_listeners:
+# -
+    ## @param port - integer - required
+    ## The UDP port to listen on for incoming trap packets.
+    #
+    # port: <PORT>
+
+    ## @param bind_host - string - optional
+    ## The hostname to listen on for incoming trap packets.
+    ## Defaults to the global `bind_host` config option value.
+    #
+    # bind_host: <BIND_HOST>
+
+    ## @param  version - string - optional
+    ## The SNMP version to use.
+    ## One of: "1" (SNMPv1), "2c" (SNMPv2c), "3" (SNMPv3).
```

Comment:
Do we really want SNMP 1 (I would say no, but who knows, some customer may have hardware from the 90s)

Reply:
It's cheap enough to support it (built into GoSNMP w/o any extra work), but I don't have any background yet on whether this is a required feature for customers.